lib/util/http/host-rules.ts

```diff
@@ -162,6 +163,18 @@ export function applyHostRules<GotOptions extends HostRulesGotOptions>(
   if (!hasProxy() && foundRules.enableHttp2 === true) {
     options.http2 = true;
   }
+
+  if (
+    foundRules.certificateAuthority !== null ||
+    foundRules.privateKey !== null ||
+    foundRules.certificate !== null
```

Comment:
This will evaluate to true if it's `undefined` or an empty string. Is that intentional?

Reply:
nah. you are right, use `is.nonEmptyString` instead.